### PR TITLE
[skipci] [Doctest] added contrib regression (``FractionalBias``, ``FractionalAbsoluteError``, ``CanberraMetric``, ``GeometricMeanAbsoluteError``)

### LINKED ISSUE #2265: [docs] Add doctests for metrics
### 📚 Documentation

Following #2230 , I listed the metrics places (suggested by @sdesrozis) where doctests can be added.

#### Metrics

- [x] Average (#2332)
- [x] GeometricAverage (#2332)
- [x] VariableAccumulation (This class is an helper, no test needed)
- [x] Accuracy (#2345)
- [x] ConfusionMatrix (#2336)
- [x] ClassificationReport (#2346)
- [x] DiceCoefficient (#2336)
- [x] JaccardIndex (#2336)
- [x] IoU (#2336)
- [x] mIoU (#2336)
- [x] EpochMetric (#2351)
- [x] Fbeta (#2340)
- [x] Loss (#2335)
- [x] MeanAbsoluteError (#2280)
- [x] MeanPairwiseDistance (#2307)
- [x] MeanSquaredError (#2280)
- [x] MetricsLambda (#2352)
- [x] MultiLabelConfusionMatrix (#2347)
- [x] Precision (#2340)
- [x] PSNR (#2311)
- [x] Recall (#2340)
- [x] RootMeanSquaredError (#2307)
- [x] RunningAverage (#2348)
- [x] SSIM (#2241)
- [x] TopKCategoricalAccuracy (#2284)
- [x] Bleu (#2317)
- [x] Rouge (#2317)
- [x] RougeL (#2317)
- [x] RougeN (#2317)
- [x] InceptionScore (#2349)
- [x] FID (#2349)

#### Contrib Metrics

- [x] AveragePrecision (#2341)
- [x] CohenKappa (#2321)
- [x] PrecisionRecallCurve (#2341)
- [x] ROC_AUC (#2341)
- [x] RocCurve (#2341)
- [x] CanberraMetric (#2323)
- [x] FractionalAbsoluteError (#2323)
- [x] FractionalBias (#2323)
- [x] GeometricMeanAbsoluteError (#2323)
- [x] GeometricMeanRelativeAbsoluteError (#2324)
- [x] ManhattanDistance (#2324)
- [x] MaximumAbsoluteError (#2324)
- [x] MeanAbsoluteRelativeError (#2324)
- [x] MeanError (#2324)
- [x] MeanNormalizedBias (#2324)
- [x] MedianAbsoluteError (#2324)
- [x] MedianAbsolutePercentageError (#2324) 
- [x] MedianRelativeAbsoluteError (#2324)
- [x] R2Score (#2324)
- [x] WaveHedgesDistance (#2324) 

This is a great issue for first comers and those who want to learn more about Sphinx. This issue is the community effort.
If you want to contribute, please comment on this issue with the metric you want to add doctest (one by one).

Example on how to add doctest, see #2241 

Thanks!

### PR DESCRIPTION
**Pull request by Ishan-Kumar2**
Addresses #2265 

Description:
Added Doctests for ``FractionalBias``, ``FractionalAbsoluteError``, ``CanberraMetric``, ``GeometricMeanAbsoluteError``

Check list:

- [ ] New tests are added (if a new feature is added)
- [X] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/contrib/metrics/regression/canberra_metric.py [hunk 1]
```diff
@@ -36,7 +36,23 @@ class CanberraMetric(_BaseRegression):
 
     .. _`Botchkarev 2018`:
         https://arxiv.org/ftp/arxiv/papers/1809/1809.03006.pdf
-
+    
+    .. testcode::
+
+            def process_function(engine, batch):
```
Comment (sdesrozis) [inline ignite/contrib/metrics/regression/canberra_metric.py:42]: Please have a look to others doctests recently introduced. You can use `default_evaluator` which is defined in https://github.com/pytorch/ignite/blob/master/docs/source/conf.py

It means your code would be simplified 
```python
metric = CanberraMetric()
metric.attach(default_evaluator, 'canberra')
preds = torch.Tensor([[3.8], [9.9], [-5.4], [2.1]])
target = preds * 1.5
state = default_evaluator.run([[preds, target]])
print(state.metrics['canberra'])
```
> 

>

File: ignite/contrib/metrics/regression/canberra_metric.py [hunk 2]
```diff
@@ -36,7 +36,23 @@ class CanberraMetric(_BaseRegression):
...
+                y_pred, y = batch
+                return y_pred, y
+            engine = Engine(process_function)
+            metric = CanberraMetric()
+            metric.attach(engine, 'canberra')
+            preds = torch.Tensor([[3.8], [9.9], [-5.4], [2.1]])
```
Comment (sdesrozis) [inline ignite/contrib/metrics/regression/canberra_metric.py:48]: I’m wondering whether `y_true` is a better name than `preds`.

```suggestion
            y_true = torch.Tensor([[3.8], [9.9], [-5.4], [2.1]])
```
and `targets` would be `y_pred`.

What do you think ? The point is we have to rename a plenty of doctests…

File: ignite/contrib/metrics/regression/fractional_absolute_error.py [hunk 3]
```diff
@@ -33,7 +33,23 @@ class FractionalAbsoluteError(_BaseRegression):
         device: specifies which device updates are accumulated on. Setting the
             metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
             non-blocking. By default, CPU.
+    
+    .. testcode::
 
+            def process_function(engine, batch):
```
Comment (sdesrozis) [inline ignite/contrib/metrics/regression/fractional_absolute_error.py:39]: Same comments than above.

File: ignite/contrib/metrics/regression/fractional_bias.py [hunk 4]
```diff
@@ -33,7 +33,23 @@ class FractionalBias(_BaseRegression):
         device: specifies which device updates are accumulated on. Setting the
             metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
             non-blocking. By default, CPU.
+	
+	.. testcode::
```
Comment (sdesrozis) [inline ignite/contrib/metrics/regression/fractional_bias.py:38]: Same comments than above.

File: ignite/contrib/metrics/regression/geometric_mean_absolute_error.py [hunk 5]
```diff
@@ -33,7 +33,22 @@ class GeometricMeanAbsoluteError(_BaseRegression):
         device: specifies which device updates are accumulated on. Setting the
             metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
             non-blocking. By default, CPU.
+    .. testcode::
 
+            def process_function(engine, batch):
```
Comment (sdesrozis) [inline ignite/contrib/metrics/regression/geometric_mean_absolute_error.py:38]: Same comments than above.

**Comment by sdesrozis** [comment]
@Ishan-Kumar2 Thank you very much for your help. I left a few comments. Note that some tests are failing because of formatting (or a similar thing).

File: ignite/contrib/metrics/regression/canberra_metric.py (hunk 1 above)
Comment (Ishan-Kumar2) [reply ignite/contrib/metrics/regression/canberra_metric.py]: Oh I missed that. I have changed to this now.

**Review by sdesrozis — APPROVED** [review]
@Ishan-Kumar2 Thanks ! LGTM

**Pull request merged**